cors/resources/cors-makeheader.py

```diff
@@ -1,67 +1,67 @@
+            if request.GET.first(b"check", None) == b"keep":
                 request.server.stash.put(token, value)
-            body = "1"
+            body = u"1"
         else:
-            body = "0"
-        return [("Content-Type", "text/plain")], body
+            body = u"0"
+        return [(b"Content-Type", b"text/plain")], body
 
 
-    if origin != 'none':
-        response.headers.set("Access-Control-Allow-Origin", origin)
-    if 'origin2' in request.GET:
-        response.headers.append("Access-Control-Allow-Origin", request.GET.first('origin2'))
+    if origin != None:
+        response.headers.set(b"Access-Control-Allow-Origin", origin)
+    if b'origin2' in request.GET:
+        response.headers.append(b"Access-Control-Allow-Origin", request.GET.first(b'origin2'))
 
     #Preflight
-    if 'headers' in request.GET:
-        response.headers.set("Access-Control-Allow-Headers", request.GET.first('headers'))
-    if 'credentials' in request.GET:
-        response.headers.set("Access-Control-Allow-Credentials", request.GET.first('credentials'))
-    if 'methods' in request.GET:
-        response.headers.set("Access-Control-Allow-Methods", request.GET.first('methods'))
+    if b'headers' in request.GET:
+        response.headers.set(b"Access-Control-Allow-Headers", request.GET.first(b'headers'))
+    if b'credentials' in request.GET:
+        response.headers.set(b"Access-Control-Allow-Credentials", request.GET.first(b'credentials'))
+    if b'methods' in request.GET:
+        response.headers.set(b"Access-Control-Allow-Methods", request.GET.first(b'methods'))
 
-    code_raw = request.GET.first('code', None)
+    code_raw = request.GET.first(b'code', None)
     if code_raw:
         code = int(code_raw)
     else:
         code = None
-    if request.method == 'OPTIONS':
+    if request.method == u'OPTIONS':
         #Override the response code if we're in a preflight and it's asked
-        if 'preflight' in request.GET:
-            code = int(request.GET.first('preflight'))
+        if b'preflight' in request.GET:
+            code = int(request.GET.first(b'preflight'))
 
         #Log that the preflight actually happened if we have an ident
-        if 'token' in request.GET:
-            request.server.stash.put(request.GET['token'], True)
+        if b'token' in request.GET:
+            request.server.stash.put(request.GET[b'token'], True)
 
-    if 'location' in request.GET:
+    if b'location' in request.GET:
         if code is None:
             code = 302
 
         if code >= 300 and code < 400:
-            response.headers.set("Location", request.GET.first('location'))
+            response.headers.set(b"Location", request.GET.first(b'location'))
 
     headers = {}
-    for name, values in request.headers.iteritems():
+    for name, values in request.headers.items():
         if len(values) == 1:
-            headers[name] = values[0]
+            headers[name.decode("iso-8859-1")] = values[0].decode("iso-8859-1")
```

Comment:
I thought we'd use abstractions to not have to mention iso-8859-1 directly?

Reply:
@ziransun you can `from wptserve.utils import isomorphic_decode` and use that instead, same as the other places.

Reply:
Updated. Thank you!